pkg/morph/client/container/load.go

```diff
@@ -53,34 +54,105 @@ type EstimationID []byte
 
 // ListLoadEstimationsByEpoch returns a list of container load estimations for to the specified epoch.
 // The list is composed through Container contract call.
-func (c *Client) ListLoadEstimationsByEpoch(epoch uint64) ([]EstimationID, error) {
+func (c *Client) ListLoadEstimationsByEpoch(epoch uint64) ([]*Estimations, error) {
 	invokePrm := client.TestInvokePrm{}
 	invokePrm.SetMethod(listSizesMethod)
 	invokePrm.SetArgs(epoch)
 
-	prms, err := c.client.TestInvoke(invokePrm)
+	kvs, err := c.client.TestInvokeIterator(invokePrm)
 	if err != nil {
 		return nil, fmt.Errorf("could not perform test invocation (%s): %w", listSizesMethod, err)
-	} else if ln := len(prms); ln != 1 {
-		return nil, fmt.Errorf("unexpected stack item count (%s): %d", listSizesMethod, ln)
 	}
 
-	prms, err = client.ArrayFromStackItem(prms[0])
-	if err != nil {
-		return nil, fmt.Errorf("could not get stack item array from stack item (%s): %w", listSizesMethod, err)
-	}
+	resMap := make(map[cid.ID]*Estimations)
+	for i := range kvs {
+		kv, err := client.ArrayFromStackItem(kvs[i])
+		if err != nil {
+			return nil, fmt.Errorf("could not unwrap iterator key-value pair: %w", err)
+		}
 
-	res := make([]EstimationID, 0, len(prms))
-	for i := range prms {
-		id, err := client.BytesFromStackItem(prms[i])
+		if len(kv) != 2 { // key-value array
+			return nil, fmt.Errorf("incorrect contract key-value struct size: %d", len(kv))
+		}
+
+		cnr, err := cidFromStorageKey(kv[0])
+		if err != nil {
+			return nil, fmt.Errorf("storage key handling: %w", err)
+		}
+
+		e, err := estimationFromStorageValue(kv[1])
 		if err != nil {
-			return nil, fmt.Errorf("could not get ID byte array from stack item (%s): %w", listSizesMethod, err)
+			return nil, fmt.Errorf("storage value handling: %w", err)
 		}
 
-		res = append(res, id)
+		if ee, found := resMap[cnr]; found {
+			ee.Values = append(ee.Values, e)
+		} else {
+			resMap[cnr] = &Estimations{
```

Comment:
And BTW, it would be nice to implement `stackitem.Convertible` interface for `Estimations` (and leave the `ToStackItem` panicing). Then you can place all this serialization-related code to `FromStackItem` method and make the main method cleaner. But it's just a suggestion.

I think we can also add `Convertible` support to the `unwrap` package.

Reply:
Ref. https://github.com/nspcc-dev/neo-go/issues/3100

Reply:
hope, i got you right, check, please. but it not gonna look perfecr because of the iteration values and `Estimation`+`Estimations` pair (there can be more that one estimation per an epoch theoretically)